dbms/src/Interpreters/ThreadStatusExt.cpp

```diff
@@ -118,6 +129,74 @@ void ThreadStatus::finalizePerformanceCounters()
     }
 }
 
+namespace {
+    void queryProfilerTimerHandler(int /* sig */, siginfo_t * /* info */, void * context) {
+        DB::WriteBufferFromFileDescriptor out(TracePipe::instance().write_fd);
```

Comment:
I doubt you can call TracePipe constructor inside signal handler.
Also for initialization of non-trivial function static variables (they use global mutex and it's non signal safe).